### ISSUE TITLE
Install dependencies from haxelib.json

### ISSUE THREAD

**Issue by JerryGreen** [HaxeFoundation/haxe#9941]
There's already a command:

```bash
haxelib install all
```

What it does: it reads current folder for `.hxml` files, lookups `-lib` flags, and install all these libs with the latest versions. Which is quite handy at first, but it has two cons:

- It lookups the _current folder_ (non-reqursively)
- It installs the _latest versions_ (which may be wrong)

And there's `haxelib.json` which lists the dependencies with actual versions.

I came from JS/TS world, and kinda expected `haxelib install` to install all versions taken from dependencies from `haxelib.json`, but it seems nor `haxelib install`, nor `haxelib install all` doesn't work this way.

I personally don't have a problem with it, since I can just list those dependencies in a file like `install.sh`:

```bash
haxelib install heaps 1.8.0
haxelib install hlsdl 1.10.0
haxelib install hldx 1.10.0
# etc
```

And run `./install.sh` in terminal. So even if some time will pass, it will be possible to setup the project with just one command, and not meet dozens of errors. But unfortunately there's no guideline or something like that, so I see some projects or examples like these:

https://github.com/Beeblerox/Simplest-Heaps-Examples

Which is quite a lot of work, but kinda useless work, because those examples just don't work with modern versions, and it's not possible to see them in action.

## My suggestion

- Make command `haxelib install` to read dependencies from `haxelib.json`, install em with specified versions
- Make commands like `haxelib install heaps` to add the latest version of heaps into `haxelib.json` (and create one, if needed)
- ❔ Probably remove `haxelib install all` ❔

**Comment by kLabz** [comment]
You can also `haxelib install install.hxml` with `install.hxml` containing specific versions:

```hxml
# from haxelib
-lib js-object:0.0.7

# from git
-lib some-lib:git:git@github.com:user/some-repo.git
# from git with reference (branch, tag, commit hash..)
-lib other-lib:git:git@github.com:user/repo.git#281f723e
```

I personnally add `--skip-dependencies` to haxelib command and add dependencies to this `install.hxml` myself to be able to pin their versions too.

See also https://github.com/HaxeFoundation/haxe/wiki/Haxe-haxec-haxelib-plan for future plans.

**Comment by JerryGreen** [comment]
Actually, that is a handy thing. I suspected something like that is possible, even tried `-lib js-object@0.0.7` (@ symbol), but it didn't work. Neither I found any info about versions in [docs](https://haxe.org/manual/compiler-usage-hxml.html).

I've just noticed it's shown in schema hint though:

![image](https://user-images.githubusercontent.com/13215662/98091559-694da500-1ea7-11eb-928c-6189de099364.png)

But what's about `--skip-dependencies`? How do you use it?

**Comment by kLabz** [comment]
It prevents haxelib from installing dependencies of libraries I install, because I cannot pin their versions (the libraries might pin their dependencies' versions but that may not be what I want either). So I add my libraries' dependencies myself in `install.hxml` too

**Comment by JerryGreen** [comment]
@kLabz yeah, I understand meaning of `install.hxml`, - that's pretty handy. You run something like `haxelib install install.hxml`, right?

I don't get where do you get benefit of using `--skip-dependencies`

**Comment by kLabz** [comment]
My previous message was entirely about `--skip-dependencies` flag

**Comment by JerryGreen** [comment]
@kLabz oh shi, it's for DEPENDENCIES of your DEPENDENCIES... I got it now. I'm ~just not very smart~ awake around 24hr already, and struggled to figure out why would you skip dependencies while you're installing dependencies.

<img width="250" height="166.5"  src="https://i.imgflip.com/5r1uy.jpg"/>
